index.rst

```diff
@@ -2789,63 +2789,94 @@ rows, rather than some finite number (thousands or less).
-the system cache and memory subsystem. This may leave some disks idle,
-but is likely to produce the maximum throughput, unless the skew in CPU
-costs among scanning queries is too great.
+to implement shared scans.  Instead, it issues query fragments ordered to
+maximize locality of access in data and time, and tries to lock the files
+associated with the tables in memory as much as possible. Using the
+identified scan tables and their ratings, the worker places them on the
+appropriate scheduler. There will be at least three schedulers. One for
+queries expected to be complete in under an hour, which are expected to
+be related to the Object table. One for queries expected to take less than
+eight hours, expected to be related to Object_Extra. And one for scans
+expected to take twelve hours for ForcedSource and/or Source tables. The
+reasoning being that a single slow query can impede the progress of a
+shared scan and all the other user queries on that scan. There may be a
+need for another scheduler to handle queries taking more than 12 hours.
+
+Each scheduler places incoming chunk queries into one of two priority
+queues sorted by chunk id then scan rating of the individual tables. If
+the query is for a chunk after the currently scanning chunk id, it is
+placed on the active priority queue, otherwise it is placed on the pending
+priority queue. After chunk id, the priority queue is sorted by the table
+with highest scan rating to ensure that the largest tables in the chunk
+are grouped together.
+
+The once the query is on the appropriate scheduler, the algorithm proceeds
+as follows. When a dispatch slot is available, it checks the highest
+priority scheduler. If that scheduler has a query fragments, hereafter
+called tasks, and it is not at its quota limit, it is allowed to start its
+next task, otherwise the worker checks the next scheduler. It continues
+doing this until a task has been started or all the schedulers have been
+checked.
+
+Each scheduler is only allowed to start a task under certain circumstances.
+There must be enough threads available from the pool so that none of the
+other schedulers are starved for threads as well as enough memory available
+to lock all the tables for the task in memory. If the scheduler has no tasks
+running, it may start one task and have memory reserved for the tables in
+that task. This should prevent any scheduler from hanging due to memory
+starvation without requiring complicated logic but could incur extra
+disk I/O. More on locking tables in memory later.
+
+Schedulers check for tasks by first checking the head of the active
+priority queue. If the active priority queue is empty, and the pending
+priority queue is not, then the active and pending queues are swapped
+with the task being taken from the top of the “new” active queue.
+
+Since the queries are being run by a separate DBMS instance of which there
+is little control of how it goes about running queries. The worker can
```

Comment:
This needs to be the same sentence, not a new one since you used the word "Since" in the beginning. "[...] running queries, the worker can [...]"